clang/lib/CIR/CodeGen/CIRGenBuiltin.cpp

```diff
@@ -199,6 +199,31 @@ RValue CIRGenFunction::emitBuiltinExpr(const GlobalDecl &gd, unsigned builtinID,
     return RValue::get(
         builder.createBitcast(allocaAddr, builder.getVoidPtrTy()));
   }
+  
+  case Builtin::BI__builtin_constant_p: { 
+    auto loc = getLoc(e->getSourceRange());
+  
+    Expr::EvalResult evalResult;
+    
+    // Try to evaluate at compile time first
+    if (e->getArg(0)->EvaluateAsRValue(evalResult, getContext()) &&
```

Comment:
You've diverged from both the incubator and classic codegen here. That's OK if your changes improve the code while keeping the same semantics, but I don't think what you've done here meets those criteria. We need this to behave exactly like classic codegen. Please use the code from the incubator.